src/components/component.ts

```diff
@@ -450,10 +450,10 @@ export class Component {
 
     if (this._isAnchored) {
       this._element.remove();
+      this._isAnchored = false;
+      this._onDetachCallbacks.callCallbacks(this);
```

Comment:
Let's loop to QE for this specific part.  While it is overhead to open a new PR, it keeps a separation of features that is very handy for testing and reviewing (note that I commented on this part in the first place, because I was confused how it fit into the whole mix), but luckily this is a decently small pull request that adding this context is not difficult.